pkg/kubelet/pluginmanager/operationexecutor/operation_generator.go

```diff
@@ -134,34 +141,31 @@ func (og *operationGenerator) GenerateRegisterPluginFunc(
 
 func (og *operationGenerator) GenerateUnregisterPluginFunc(
 	socketPath string,
-	pluginHandlers map[string]cache.PluginHandler,
+	pathToHandlers *cache.MapHandlerWithName,
 	actualStateOfWorldUpdater ActualStateOfWorldUpdater) func() error {
 
 	unregisterPluginFunc := func() error {
-		client, conn, err := dial(socketPath, dialTimeoutDuration)
-		if err != nil {
-			return fmt.Errorf("UnregisterPlugin error -- dial failed at socket %s, err: %v", socketPath, err)
-		}
-		defer conn.Close()
-
-		ctx, cancel := context.WithTimeout(context.Background(), time.Second)
-		defer cancel()
-
-		infoResp, err := client.GetInfo(ctx, &registerapi.InfoRequest{})
-		if err != nil {
-			return fmt.Errorf("UnregisterPlugin error -- failed to get plugin info using RPC GetInfo at socket %s, err: %v", socketPath, err)
+		var pair cache.HandlerWithName
+		handlerFound := false
+		pathToHandlers.Lock()
+		if len(pathToHandlers.Handlers) != 0 {
+			pair, handlerFound = pathToHandlers.Handlers[socketPath]
 		}
-
-		handler, ok := pluginHandlers[infoResp.Type]
-		if !ok {
-			return fmt.Errorf("UnregisterPlugin error -- no handler registered for plugin type: %s at socket %s", infoResp.Type, socketPath)
+		pathToHandlers.Unlock()
+
+		if handlerFound {
```

Comment:
nit: idiomatic go style we could do `if !handlerFound { error}` then the "happy logic" doesn't need to be nested